tests/AppSwitcher.test.js

```diff
@@ -0,0 +1,234 @@
+            },
+          ],
+        }}
+      />
+      {children}
+    </Router>
+  );
+};
+
+const neetoApps = ["KB", "Desk", "Planner"];
+
+describe("AppSwitcher", () => {
+  it("should render without error", () => {
+    const { getByText } = render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="KB"
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    for (const app of neetoApps) {
+      expect(getByText(app)).toBeInTheDocument();
+    }
+  });
+
+  it("should not render when isOpen is false", () => {
+    const { queryByText } = render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen={false}
+          neetoApps={neetoApps}
+          activeApp="KB"
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    for (const app of neetoApps) {
+      expect(queryByText(app)).not.toBeInTheDocument();
+    }
+  });
+
+  it("should render active app correctly", () => {
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+    const neetoUIClassPrefix = "neeto-ui-app-switcher-link--";
+    const activeLink = document.querySelector(`.${neetoUIClassPrefix}active`);
+    expect(activeLink).toBeInTheDocument();
+    expect(getByText(activeLink, "Planner")).toBeInTheDocument();
+  });
+
+  it("should render different sizes correctly", () => {
+    const size = "lg";
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          size={size}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    const neetoUIClassPrefix = "neeto-ui-app-switcher__wrapper-size--";
+    expect(
+      document.querySelector(`.${neetoUIClassPrefix}${size}`)
+    ).toBeInTheDocument();
+  });
+
+  it("should render className correctly", () => {
+    const size = "lg";
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          className="test-class"
+          size={size}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    expect(document.querySelector(".test-class")).toBeInTheDocument();
+  });
+
+  it("should close when the back button is pressed", () => {
+    const onClose = jest.fn();
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          onClose={onClose}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    const backButton = document.querySelector(
+      "button[data-cy='app-switcher-back-button']"
+    );
```

Comment:
Back button can be found by text. Do not use `document` or `querySelector` in these cases.

```suggestion
    const backButton = screen.getByText("Back");
```